server/src/main/java/com/broadcom/lsp/cobol/service/utils/CustomThreadPoolExecutorService.java

```diff
@@ -0,0 +1,94 @@
+package com.broadcom.lsp.cobol.service.utils;
+
+import com.google.inject.Inject;
+import com.google.inject.Singleton;
+import com.google.inject.name.Named;
+import lombok.NonNull;
+
+import java.util.Map;
+import java.util.concurrent.*;
+
+import static java.util.Collections.singletonMap;
+
+/**
+ * {@inheritDoc}
+ *
+ * <p>Implements API for our CustomThreadPoolExecutor created using ThreadPoolExecutor and
+ * ScheduledThreadPoolExecutor.
+ *
+ * <p>The main configuration parameters are: corePoolSize, maximumPoolSize, and keepAliveTime. The
+ * pool consists of a fixed number of core threads that are kept inside all the time, and some
+ * excessive threads that may be spawned and then terminated when they are not needed anymore.
+ */
+@Singleton
+public class CustomThreadPoolExecutorService implements CustomThreadPoolExecutor {
```

Comment:
The name doesn't reflect the purpose of the class. It is either a Provider that just holds and returns the executors or an Executor, but in this case, it should not expose the nested executors and just provide the API to control the tasks.

Reply:
Changed, will rebase and squash